app/components/Calendar.tsx

```diff
@@ -1,89 +1,62 @@
-import dayjs from 'dayjs';
-import { useState, useCallback, useEffect } from 'react';
+import React from 'react';
 import { View } from 'react-big-calendar';
-import { CalendarEventProps, CalEvent } from '~/utils/interfaces';
-import EventModal from '~/components/common/eventModal';
+import dayjs from 'dayjs';
+import axios from 'axios';
+import { CalEvent } from '~/utils/interfaces';
 import RdsCalendar from '~/components/common/rdsCalendar';
 import { useStore } from '~/store/useStore';
+import { parseEvents } from '~/utils/event.utils';
+import { patchEvent } from '~/constants/urls.constants';
 
 interface CalendarProps {
   view?: View;
+  events: CalEvent[];
 }
 
-const Calendar = ({ view }: CalendarProps) => {
-  const events = useStore((state: { events: any }) => state.events);
-
-  const [eventsList, setEventsList] = useState<CalEvent[]>(events);
-  const [calendarEvent, setCalendarEvent] = useState<CalendarEventProps>({
-    event: events[0],
-    show: false,
-    new: false,
-  });
+const Calendar = ({ view, events }: CalendarProps) => {
+  const { updateEvent } = useStore((state) => state);
 
-  useEffect(() => {
-    setEventsList(events);
-  }, [events]);
+  const updateEventStateFromCalendar = async (event: CalEvent) => {
+    const { id } = event;
+    const payload = {
+      name: event.title,
+      startTime: dayjs(event?.start).valueOf(),
+      endTime: dayjs(event?.end).valueOf(),
+      location: event.location,
+      description: event.description,
+      attendees: event?.attendees ? event.attendees.map(({ attendee }) => attendee.email) : [],
+    };
```

Comment:
we can create a parser function for converting `frontend obj format` to `backend obj format` like we did for frontend to backend

Reply:
yes creating for payload